tests/helpers/perf_run_helper.rb

```diff
@@ -911,9 +913,53 @@ def get_perf_result
       puts
     end
 
+    save_average_compile_time("#{@archive_root}/avg_compile_time.txt", database)
+
     [perf, GatlingResult.new(gatling_assertions, mean_response_time)]
   end
 
+  def save_average_compile_time(file, puppetdb_host)
+    # Convert epoch values (which are created based on local time) to UTC time
+    # strings compatible with puppetdb queries.
+    start_time = Time.at(@start_epoch.to_i).getgm.strftime "%Y-%m-%d %H:%M:%S"
+    end_time = Time.at(@end_epoch.to_i).getgm.strftime "%Y-%m-%d %H:%M:%S"
+
+    query = <<~QUERY
+      query=["from", "reports",
+              ["extract", "metrics",
+                ["and",
+                  [">=", "start_time", "#{start_time}"],
+                  ["<=", "end_time", "#{end_time}"]]]]
+    QUERY
+    query = query.strip.gsub(/\n+/, " ")
+    query = query.gsub(/\s+/, " ")
+
+    curl = %W[
+      curl
+      -X GET
+      http://localhost:8080/pdb/query/v4
+      -d '#{query}'
+    ].join(" ")
+
+    result = nil
+    begin
+      bkr_res = on(puppetdb_host, curl)
+      data = JSON.parse(bkr_res.stdout.chomp)
+      data.delete_if { |report| report["metrics"]["data"].empty? }
+      config_retrieval_times = data.map do |report|
+        report["metrics"]["data"].select do |md|
+          md["category"] == "time" && (md["name"] == "config_retrieval" || md["name"] == "total")
+        end.first.fetch("value")
+      end
+      avg_config_retrieval_time = config_retrieval_times.reduce(0.0, :+) / config_retrieval_times.size
+      result = avg_config_retrieval_time.ceil
+    rescue StandardError # rubocop:disable Lint/HandleExceptions
+      # empty file will be written
```

Comment:
could we write the error to the file instead of empty?

Reply:
We could, however my bias is that doing so breaks the interface contract for the file (contains average compile time).  The absence of a value indicates that something when sideways.  I would counter propose that we emit the error via the logger instead to assist in determining why things went sideways.  Thoughts?

Reply:
My only concern with that is so many errors that get logged are lost in the mass of output that we get.  And I am not sure an error breaks the contract more then a nil value...  All in all I would rather have some kind of error file that errors are logged to instead of the mass of output we get from the run...

Reply:
I have pushed an update to write a JSON payload to the file.  If an error is encountered, it will be written as part of the JSON data structure.